edb/server/server.py

```diff
@@ -1703,17 +1704,41 @@ async def _resolve_localhost() -> List[str]:
 def _fix_wildcard_host(hosts: Sequence[str]) -> Sequence[str]:
     # Even though it is sometimes not a conflict to bind on the same port of
     # both the wildcard host 0.0.0.0 and some specific host at the same time,
-    # we're still discarding other hosts if 0.0.0.0 is present because it
-    # should behave the same and we could avoid potential conflicts.
+    # we're still discarding other hosts if 0.0.0.0 (or ::0) is present
+    # because it should behave the same and we could avoid potential conflicts.
 
-    if '0.0.0.0' in hosts:
-        if len(hosts) > 1:
+    ip_hosts = set()
+    for host in hosts:
+        try:
+            ip = ipaddress.ip_address(host)
+        except ValueError:
+            pass
+        else:
+            ip_hosts.add(ip)
+
+    if not ip_hosts:
+        return hosts
+
+    ipv4_wc = ipaddress.ip_address('0.0.0.0')
+    ipv6_wc = ipaddress.ip_address('::')
+
+    wc_hosts = []
+
+    if ipv4_wc in ip_hosts:
+        wc_hosts.append('0.0.0.0')
+
+    if ipv6_wc in ip_hosts:
+        wc_hosts.append('::')
+
+    if wc_hosts:
```

Comment:
I think you have to manage the IPv4 and IPv6 sets separately here.  Otherwise you'll discard IPv4 hosts if I specified (`127.0.0.1, ::`).

Reply:
Right, fixed